### ISSUE TITLE
Rarely exception on PopPopupaAsync

### ISSUE THREAD

**Issue by Hugoxz** [rotorgames/Rg.Plugins.Popup#84]
Hi @rotorgames !

I did some tests of memory consuption and aparently you improved the dispose right?
But some times at `await Navigation.PopPopupAsync();` throws exception `System.IndexOutOfRangeException: There is not page in PopupStack
  at Rg.Plugins.Popup.Services.PopupNavigation.PopAsync (System.Boolean animate) [0x0000c] in <e4a0d036d16d43a8a07fbef673df812b>:0 
  at Rg.Plugins.Popup.Extensions.NavigationExtension.PopPopupAsync (Xamarin.Forms.INavigation sender, System.Boolean animate) [0x00000] in <e4a0d036d16d43a8a07fbef673df812b>:0 `.

In my tests I open and close many popups one by one the fastest as possible. The exception varies after 15, 30, 50 open-close popup.
Currently I using try-catch to my app dont crash and works fine.

Thank you.

**Comment by rotorgames** [comment]
@Hugoxz Hi. Did you do it with await operators? I see that you pop page but there is not page in`PopupStack` in this moment. Maybe you didn't wait for end of `PopPopupAsync`?

**Comment by Hugoxz** [comment]
I am opening the popup when get focused on an entry, here is an example:

`public class CustomEntry : Entry
    {
        public CustomEntry()
        {
            this.Focused += OnFocused;
        }

        private async void OnFocused(object sender, FocusEventArgs focusEventArgs)
        {
            this.Focused -= OnFocused;
            this.Unfocus();

            var popup = new MyPopup();
            popup.Closed += (o, args) =>
            {
                this.Focused += OnFocused;
            };

            await Navigation.PushPopupAsync(popup);
        }
    }`

`public partial class MyPopup : PopupPage
    {
        public event EventHandler Closed;

        public MyPopup()
        {
            InitializeComponent();
        }

        private async void Button_OnClicked(object sender, EventArgs e)
        {
            await Navigation.PopPopupAsync();
        }

        protected override void OnDisappearing()
        {
            base.OnDisappearing();
            Closed?.Invoke(this, EventArgs.Empty);
        }
    }`

I think the error is because another popup is trying to opening on focus before OnDisappearing is complete. Well, I think it is not the most appropriate to open the popup but with the try-catch works fine, lol.

**Comment by codebeaulieu** [comment]
I had this issue as well. For me, the issue was because I was accidentally binding my call to `PopAsync` with a boolean using ReactiveUI. I forgot to place a check in my call to ensure that the boolean was true.

```
this.WhenAnyValue(x => x.ViewModel.Postcomplete)
    .Subscribe(async val => await PopupNavigation.PopAsync(true))
    .DisposeWith(ControlBindings);
```

So the moment I loaded the PopUp dialog, a call to PopAsync was called which caused the crash. Once I included a check in my ReactiveUI code, everything worked fine.

````
this.WhenAnyValue(x => x.ViewModel.Postcomplete)
    .Where(boolVal => boolVal) // make sure is true
    .Subscribe(async val => await PopupNavigation.PopAsync(true))
    .DisposeWith(ControlBindings);
````

Just make sure your PopUp is in the stack before calling PopAsync

**Comment by rotorgames** [comment]
@Hugoxz @codebeaulieu The Rg.Plugins.Popup has blockers for repeated PopAsync calls for one page but the Rg.Plugins.Popup will throw `IndexOutOfRangeException` if there are not pages in the `PopupStack` when the `PopAsync` will be invoked.

**Comment by rotorgames** [comment]
This issue was fixed in v1.1.0-pre1. Please test it and tell me about your tests. Thank you.
